File: sdks/java/basys.sdk/src/main/java/org/eclipse/basyx/submodel/metamodel/connected/submodelelement/property/file/ConnectedFile.java
package org.eclipse.basyx.submodel.metamodel.connected.submodelelement.property.file;

import org.eclipse.basyx.submodel.metamodel.api.submodelelement.property.file.IFile;
import org.eclipse.basyx.submodel.metamodel.connected.submodelelement.ConnectedDataElement;
import org.eclipse.basyx.submodel.metamodel.map.submodelelement.property.SingleProperty;
import org.eclipse.basyx.submodel.metamodel.map.submodelelement.property.blob.Blob;
import org.eclipse.basyx.vab.modelprovider.VABElementProxy;
/**
 * "Connected" implementation of IFile
 * @author rajashek
 *
 */
public class ConnectedFile extends ConnectedDataElement implements IFile {
	public ConnectedFile(VABElementProxy proxy) {
		super(proxy);		
	}
	
	@Override
	public void setValue(String value) {
		getProxy().setModelPropertyValue(SingleProperty.VALUE, value);
		
	}

	@Override
	public String getValue() {
		return (String) getProxy().getModelPropertyValue(SingleProperty.VALUE);
	}

	@Override
	public void setMimeType(String mimeType) {
		getProxy().setModelPropertyValue(Blob.MIMETYPE, mimeType);
		
	}

	@Override
	public String getMimeType() {
		return (String) getProxy().getModelPropertyValue(Blob.MIMETYPE);
	}

}

File: sdks/c++/basys.sdk.cc/src/submodel/submodel/connected/submodelelement/property/file/ConnectedFile.cpp
/*
 * ConnectedFile.cpp
 *
 *      Author: wendel
 */

#include <BaSyx/submodel/connected/submodelelement/property/file/ConnectedFile.h>
#include <BaSyx/submodel/api/submodelelement/property/IProperty.h>
#include <BaSyx/submodel/api/submodelelement/property/blob/IBlob.h>

namespace basyx {
namespace submodel {

ConnectedFile::ConnectedFile(std::shared_ptr<vab::core::proxy::IVABElementProxy> proxy) :
  ConnectedDataElement(proxy)
{}

void ConnectedFile::setValue(const std::string & value)
{
	this->setProxyValue(IProperty::Path::Value, value);
}

const std::string & ConnectedFile::getValue() const
{
	return this->getProxyValue(IProperty::Path::Value);
}

void ConnectedFile::setMimeType(const std::string & mimeType)
{
	this->setProxyValue(IBlob::Path::MIMEType, mimeType);
}

const std::string & ConnectedFile::getMimeType() const
{
	return this->getProxyValue(IBlob::Path::MIMEType);
}

}
}
